### PR TITLE
Fix ruff low hanging fruit

### PR DESCRIPTION
**Pull request by marcelveldt**
Fix several lint issue sand optimizations based.
Mostly import order and new-style type notations.

### CODE REVIEW

File: matter_server/client/client.py [hunk 1]
```diff
@@ -48,6 +46,10 @@
 )
 
 if TYPE_CHECKING:
+    from collections.abc import Callable
```
Comment (MartinHjelmare) [inline matter_server/client/client.py:49]: I don't like this rule, that names that are only used for type checking should be imported under `if TYPE_CHECKING` flag. I think that's just annoying. I don't think there's big performance problem.

https://docs.astral.sh/ruff/rules/typing-only-standard-library-import/

File: matter_server/client/client.py [hunk 2]
```diff
@@ -125,7 +121,8 @@ def get_node(self, node_id: int) -> MatterNode:
         """Return Matter node by id or None if no node exists by that id."""
         if node := self._nodes.get(node_id):
             return node
-        raise NodeNotExists(f"Node {node_id} does not exist or is not yet interviewed")
+        msg = f"Node {node_id} does not exist or is not yet interviewed"
+        raise NodeNotExists(msg)
```
Comment (MartinHjelmare) [inline matter_server/client/client.py:125]: I don't understand what makes it less readable in the traceback:
https://docs.astral.sh/ruff/rules/f-string-in-exception/

Not sure about this one.

**Comment by marcelveldt** [comment]
This is not working as the linter config differs

File: matter_server/client/client.py [hunk 3]
```diff
@@ -4,23 +4,20 @@
 
 import asyncio
 import logging
-from types import TracebackType
-from typing import TYPE_CHECKING, Any, Callable, Dict, Final, Optional, cast
 import uuid
+from typing import TYPE_CHECKING, Any, Final, cast
```
Comment (MartinHjelmare) [inline matter_server/client/client.py:8]: I prefer having the imports sorted :abcd: regardless if they are imported with `import ...` or with `from ... import`. Can we adjust that rule? We have this sorted like that in core.

File: matter_server/server/client_handler.py [hunk 4]
```diff
@@ -229,7 +231,7 @@ def _send_message(self, message: MessageType) -> None:
         try:
             self._to_write.put_nowait(_message)
         except asyncio.QueueFull:
-            self._logger.error(
+            self._logger.exception(
```
Comment (MartinHjelmare) [inline matter_server/server/client_handler.py:234]: Which rule is this?

File: scripts/beautify_diagnostics.py [hunk 5]
```diff
@@ -2,33 +2,29 @@
...
         sys.exit(1)
 
     with open(sys.argv[1]) as f:
         data = json.load(f)
 
-    if "node" in data["data"]:
-        nodes = [data["data"]["node"]]
-    else:
-        nodes = data["data"]["server"]["nodes"]
+    nodes = [data["data"]["node"]] if "node" in data["data"] else data["data"]["server"]["nodes"]
```
Comment (MartinHjelmare) [inline scripts/beautify_diagnostics.py:19]: This creates a line that is longer than 88 characters. I don't like that.

File: scripts/generate_devices.py [hunk 6]
```diff
@@ -83,21 +84,16 @@ def __hash__(self) -> int:
         name = device["typeName"]
 
         if not name.startswith("Matter "):
-            print("Unexpected: device doesn't start with Matter. Skipping")
```
Comment (MartinHjelmare) [inline scripts/generate_devices.py:86]: We don't need to remove prints from scripts if we think the prints are useful.

File: matter_server/client/client.py (hunk 2 above)
Comment (marcelveldt) [reply matter_server/client/client.py:125]: yeah I was surprised by it as well - we can disable this one if you want

File: matter_server/client/client.py (hunk 3 above)
Comment (marcelveldt) [reply matter_server/client/client.py:8]: @frenck do you know how ?

File: scripts/beautify_diagnostics.py (hunk 5 above)
Comment (marcelveldt) [reply scripts/beautify_diagnostics.py:19]: I have set the minimal line length to 100 as 88 is really short

File: scripts/generate_devices.py (hunk 6 above)
Comment (marcelveldt) [reply scripts/generate_devices.py:86]: yeah scripts/examples may have prints

File: matter_server/server/client_handler.py (hunk 4 above)
Comment (marcelveldt) [reply matter_server/server/client_handler.py:234]: I don't know - its very hard to track

Comment (MartinHjelmare) [reply matter_server/server/client_handler.py:234]: I think it's this one:
https://docs.astral.sh/ruff/rules/error-instead-of-exception/

That's very subjective I'd say. Not sure that we should follow that one all the time.

File: matter_server/client/client.py (hunk 1 above)
Comment (marcelveldt) [reply matter_server/client/client.py:49]: We can disable it if we want ?

File: scripts/beautify_diagnostics.py (hunk 5 above)
Comment (MartinHjelmare) [reply scripts/beautify_diagnostics.py:19]: I'd rather stick with 88 characters per line. That's what we have for all projects now. It also fits nicely on mobile in landscape.

File: matter_server/client/client.py (hunk 1 above)
Comment (marcelveldt) [reply matter_server/client/client.py:49]: @frenck @MartinHjelmare do we want this rule to be disabled by default ?

File: matter_server/client/client.py (hunk 2 above)
Comment (marcelveldt) [reply matter_server/client/client.py:125]: @frenck @MartinHjelmare do we want this rule to be disabled by default ?

File: matter_server/server/client_handler.py (hunk 4 above)
Comment (marcelveldt) [reply matter_server/server/client_handler.py:234]: I've disabled this rule in the config

File: matter_server/client/client.py (hunk 2 above)
Comment (MartinHjelmare) [reply matter_server/client/client.py:125]: Personally I don't think this rule improves things.

Comment (marcelveldt) [reply matter_server/client/client.py:125]: Me too, I'm in favor as well to just disable it

File: matter_server/client/client.py (hunk 1 above)
Comment (marcelveldt) [reply matter_server/client/client.py:49]: disabled in #567

**Pull request closed without merging**